x-pack/plugins/siem/server/lib/detection_engine/signals/filter_events_with_list.test.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import uuid from 'uuid';
+import { filterEventsAgainstList } from './filter_events_with_list';
+import { mockLogger, repeatedSearchResultsWithSortId } from './__mocks__/es_results';
+
+import { ListClient } from '../../../../../lists/server';
+import { ListItemArraySchema } from '../../../../../lists/common/schemas';
+
+describe('filterEventsAgainstList', () => {
+  const someGuids = Array.from({ length: 13 }).map((x) => uuid.v4());
+
+  it('should respond with eventSearchResult if exceptionList is empty', async () => {
+    const res = await filterEventsAgainstList({
+      logger: mockLogger,
+      listClient: ({
+        getListItemByValues: async () => ([] as unknown) as ListItemArraySchema,
+      } as unknown) as ListClient,
+      exceptionsList: undefined,
+      eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3), [
+        '1.1.1.1',
+        '2.2.2.2',
+        '3.3.3.3',
+        '7.7.7.7',
+      ]),
+    });
+    expect(res.hits.hits.length).toEqual(4);
+  });
+
+  it('should throw an error if malformed exception list present', async () => {
+    try {
+      await filterEventsAgainstList({
+        logger: mockLogger,
+        listClient: ({
+          getListItemByValues: async () => ([] as unknown) as ListItemArraySchema,
```

Comment:
nit: extra cast not needed:

```ts
getListItemByValues: async () => [],
```

Should be the same with all of these with the empty arrays for the rest of this file.